bigbluebutton-html5/imports/ui/components/connection-status/queries.tsx

```diff
@@ -0,0 +1,104 @@
+import { gql } from '@apollo/client';
+
+export interface ConnectionStatusHistory {
+  statusUpdatedAt: string;
+  status: string;
+  networkRttInMs: number;
+}
+
+export interface ConnectionStatusResponse {
+  user_connectionStatusHistory: ConnectionStatusHistory[];
+}
+
+export interface ConnectionStatusSummary {
+  statusCriticalLastOccurrenceAt: string | null; // ISO timestamp or null
+  statusCriticalOccurrences: number;
+  statusDangerLastOccurrenceAt: string | null;
+  statusDangerOccurrences: number;
+  statusNormalLastOccurrenceAt: string | null;
+  statusNormalOccurrences: number;
+  statusWarningLastOccurrenceAt: string | null;
+  statusWarningOccurrences: number;
+  totalOfOccurrences: number;
+}
+
+export interface ConnectionStatusSummaryResponse {
+  user_connectionStatusSummary: ConnectionStatusSummary[];
+}
+
+export const CONNECTION_STATUS_REPORT_SUBSCRIPTION = gql`subscription ConnStatusReport {
+  user_connectionStatusReport(
+  where: {
+    _or: [
+            { clientNotResponding: { _eq: true } }, 
+            { lastUnstableStatus: { _is_null: false } }
+          ]
+  },
+  order_by: {lastUnstableStatusAt: desc}
+  ) {
+    user {
+      userId
+      name
+      avatar
+      color
+      isModerator
+      currentlyInMeeting
+    }
+    clientNotResponding
+    lastUnstableStatus
+    lastUnstableStatusAt
+    currentStatus
+    connectionAliveAt
+  }
+}`;
+
+export const USER_CURRENT_STATUS_SUBSCRIPTION = gql`
+  subscription CurrentUserConnStatus($userId: String!) {
+    user_connectionStatusReport(
+      where: {
+        user: {
+          userId: { _eq: $userId }
+        }
+      }
+    ) {
+      currentStatus
+    }
+  }
+`;
+
+export const CONNECTION_STATUS_SUBSCRIPTION = gql`subscription ConnStatus {
+  user_connectionStatus {
+    connectionAliveAt
+    userClientResponseAt
+    status
+    statusUpdatedAt
+  }
+}`;
+
+export const CONNECTION_STATUS_HISTORY = gql`
+  subscription ConnectionStatus($userId: String!) {
+    user_connectionStatusHistory(order_by: {statusUpdatedAt: asc}, where: {userId: {_eq: $userId}}) {
```

Comment:
I'd suggest ordering the status history in descending order so that users see newest statuses first.